catalog/dags/common/elasticsearch.py

```diff
@@ -0,0 +1,223 @@
+    # have already ensured in a previous check that there is exactly one value in the
+    # response.
+    config = next(iter(response.values()))
+    return config
+
+
+def remove_excluded_index_settings(index_config):
+    """
+    Remove fields from the given index configuration that should not be included when
+    using it to create a new index.
+    """
+    # Remove fields from the current_index_config that should not be copied
+    # over into the new index (such as uuid)
+    for setting in EXCLUDED_INDEX_SETTINGS:
+        index_config.get("settings", {}).get("index", {}).pop(setting)
+
+    # Aliases should also not by applied automatically
+    index_config.pop("aliases")
+
+    return index_config
+
+
+@task
+def create_index(index_config, es_host: str):
+    es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+
+    new_index = es_conn.indices.create(**index_config)
+
+    return new_index
+
+
+@task_group(group_id="trigger_and_wait_for_reindex")
+def trigger_and_wait_for_reindex(
+    destination_index: str,
+    source_index: str,
+    query: dict,
+    timeout: timedelta,
+    requests_per_second: int,
+    es_host: str,
+    max_docs: int | None = None,
+    slices: Union[int, Literal["auto"]] = "auto",
+):
+    @task
+    def trigger_reindex(
+        es_host: str,
+        destination_index: str,
+        source_index: str,
+        query: dict,
+        requests_per_second: int,
+        max_docs: int | None,
+        slices: Union[int, Literal["auto"]],
+    ):
+        es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+        source = {"index": source_index}
+        # An empty query is not accepted; only pass it
+        # if a query was actually supplied
+        if query:
+            source["query"] = query
+
+        response = es_conn.reindex(
+            source=source,
+            dest={"index": destination_index},
+            max_docs=max_docs,
+            # Parallelize indexing when not None
+            slices=slices,
+            # Do not hold the slot while awaiting completion
+            wait_for_completion=False,
+            # Immediately refresh the index after completion to make
+            # the data available for search
+            refresh=True,
+            # Throttle
+            requests_per_second=requests_per_second,
+        )
+        return response["task"]
+
+    def _wait_for_reindex(task_id: str, expected_docs: int, es_host: str):
+        es_conn = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+
+        response = es_conn.tasks.get(task_id=task_id)
+
+        count = response.get("task", {}).get("status", {}).get("total")
+        if expected_docs and count != expected_docs:
+            logger.info(
+                f"Reindexed {count} documents, but {expected_docs} were expected."
+            )
+        else:
+            logger.info(f"Reindexed {count} documents.")
+        return response.get("completed")
```

Comment:
FWIW it looks like we can use `@task.sensor` here for this instead of `PythonSensor`! https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/python.html#pythonsensor

Might be useful to make this a reusable function too? Your call 😄